lib/Conversion/TorchToTcp/Misc.cpp

```diff
@@ -114,11 +116,14 @@ class ConvertAtenBroadcastLikeOps : public OpConversionPattern<AtenOpT> {
       // Note: The order of checks in this boolean expression matters!
       if (isNewDim || isDynamicDim ||
           (!isDimSizePreserved && doesDimSizeChange)) {
-        axes.push_back(i);
-        newDimSize = rewriter.create<torch::TorchConversion::ToI64Op>(
-            op->getLoc(), newDimSize);
-        resultShape.push_back(rewriter.create<arith::IndexCastOp>(
-            op->getLoc(), rewriter.getIndexType(), newDimSize));
+        // only broadcast if the input dim size is 1
+        if (newInputShape[i] == 1) {
```

Comment:
This might be a little nicer with two booleans something like
```diff
+bool inputDimBroadcastable = newInputShape[i] == 1);
+// Note: The order of checks in this boolean expression matters!
+bool outputDimBroadcasted = isNewDim || isDynamicDim ||
+          (!isDimSizePreserved && doesDimSizeChange);
-// Note: The order of checks in this boolean expression matters!
-if (isNewDim || isDynamicDim ||
-          (!isDimSizePreserved && doesDimSizeChange)) {
+if (outputDimBroadcasted && inputDimBroadcastable) {

Reply:
Yup I like this idea. Will update it.